### ISSUE TITLE
[BUG] Azure.AI.TextAnalytics.AnalyzeActionsOperation throws System.NullReferenceException

### ISSUE THREAD

**Issue by HenryvanderVegte** [Azure/azure-sdk-for-net#24692]
**Describe the bug**
I am trying to pick up a previously created Text Analytics job using the async API through the 

```csharp
public AnalyzeActionsOperation(string operationId, TextAnalyticsClient client);
```

constructor. When calling something like GetValuesAsync() on that operation, the call throws a NullReferenceException.
However, I can see that the job succeeded when doing a HTTP GET on ..../text/analytics/v3.1/analyze/jobs/{operationId}

**Expected behavior**
I would expect that I can pick up the results from my Async Operation using the operationId. In my scenario, I can't keep the AnalyzeActionsOperation that the StartAnalyzeActionsAsync() call returns alive, but need to pick up the result independent of the job creation.

**Actual behavior (include Exception or Stack Trace)**
Calls like WaitForCompletionAsync or GetValuesAsync throw a NullReferenceException:

```
System.NullReferenceException
  HResult=0x80004003
  Message=Object reference not set to an instance of an object.
  Source=Azure.AI.TextAnalytics
  StackTrace:
   at Azure.AI.TextAnalytics.Transforms.<>c__DisplayClass39_0`1.<SortHeterogeneousCollection>b__0(T result)
   at System.Linq.EnumerableSorter`2.ComputeKeys(TElement[] elements, Int32 count)
   at System.Linq.EnumerableSorter`1.ComputeMap(TElement[] elements, Int32 count)
   at System.Linq.EnumerableSorter`1.Sort(TElement[] elements, Int32 count)
   at System.Linq.OrderedEnumerable`1.ToList()
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Azure.AI.TextAnalytics.Transforms.SortHeterogeneousCollection[T](List`1 collection, IDictionary`2 idToIndexMap)
   at Azure.AI.TextAnalytics.Transforms.ConvertToAnalyzeSentimentResultCollection(SentimentResponse results, IDictionary`2 idToIndexMap)
   at Azure.AI.TextAnalytics.Transforms.ConvertToAnalyzeSentimentActionsResults(AnalyzeJobState jobState, IDictionary`2 idToIndexMap, IDictionary`2 tasksErrors)
   at Azure.AI.TextAnalytics.Transforms.ConvertToAnalyzeActionsResult(AnalyzeJobState jobState, IDictionary`2 map)
   at Azure.AI.TextAnalytics.AnalyzeActionsOperation.<Azure-Core-IOperation<Azure-AsyncPageable<Azure-AI-TextAnalytics-AnalyzeActionsResult>>-UpdateStateAsync>d__55.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd(Task task)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1.ConfiguredValueTaskAwaiter.GetResult()
   at Azure.Core.OperationInternal`1.<UpdateStateAsync>d__35.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Azure.Core.OperationInternal`1.<UpdateStatusAsync>d__34.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1.ConfiguredValueTaskAwaiter.GetResult()
   at Azure.Core.OperationInternal`1.<UpdateStatusAsync>d__29.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1.ConfiguredValueTaskAwaiter.GetResult()
   at Azure.AI.TextAnalytics.AnalyzeActionsOperation.<UpdateStatusAsync>d__49.MoveNext()
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Threading.Tasks.ValueTask`1.get_Result()
   at System.Runtime.CompilerServices.ConfiguredValueTaskAwaitable`1.ConfiguredValueTaskAwaiter.GetResult()

```

**To Reproduce**
Running the following code reproduces the issue:

```csharp
            var textAnalyticsClient = new TextAnalyticsClient(new Uri("https://westeurope.api.cognitive.microsoft.com"), new AzureKeyCredential(subscriptionKey));

            var actions = new TextAnalyticsActions
            {
                DisplayName = "TestName",
                AnalyzeSentimentActions = new List<AnalyzeSentimentAction>() { new AnalyzeSentimentAction() }
            };
            var documents = new[] { new TextDocumentInput("testId", "This is a test.") };

            var operation = await textAnalyticsClient.StartAnalyzeActionsAsync(documents, actions).ConfigureAwait(false);

            // this call works fine
            await operation.WaitForCompletionAsync();

            var newOperation = new AnalyzeActionsOperation(operation.Id, textAnalyticsClient);

            // this call fails
            await newOperation.WaitForCompletionAsync().ConfigureAwait(false);
```

**Environment:**
 - Package: Azure.AI.TextAnalytics 5.1.0
 
 I'm running on .NET Core 3.1.20

Thanks!

**Comment by maririos** [comment]
@HenryvanderVegte  thank you so much for reporting this. I confirm I am able to reproduce the issue and understand the problem. 

What is happening is that we are trying to organize the output with the input of the documents, but when rehydrating an `AnalyzeActionsOperation` we don't know what the output was, so our code fails.

I have created a quick fix and was wondering if you could give it a try and let me know if this solves your issue. 
Here is the package: https://dev.azure.com/azure-sdk/public/_packaging?_a=package&feed=azure-sdk-for-net&package=Azure.AI.TextAnalytics&protocolType=NuGet&version=5.2.0-alpha.20211005.2&view=overview

The concern with the fix provided here, is that we won't be able to guarantee that a document and its result will have the same index in the input and result collections. 
How will this impact you?

**Comment by maririos** [comment]
Thanks @HenryvanderVegte . We are looking into other strategies that will also preserve the input order.
Once we have a package for you to test I will let you know.

Apologies for the inconveniences this is causing you

**Comment by HenryvanderVegte** [comment]
okay, thanks - looking forward!

**Comment by kinelski** [comment]
Hello @HenryvanderVegte,

We have released version 5.1.1 of Azure.AI.TextAnalytics today, and we believe it fixes the issue you reported. Could you give it a try? Please be aware that **operation IDs created with version 5.1.0 or below of the SDK still won't work**. Only long-running operations started with 5.1.1 can be recovered.

**Comment by HenryvanderVegte** [comment]
Hi @kinelski,

sorry for the delayed response. I can confirm now that this works, thanks a lot for your help!
